templates/intro-workspaces/README.ipynb

```diff
@@ -0,0 +1,334 @@
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!aws s3 cp $ANYSCALE_ARTIFACT_STORAGE/persisted_object.txt /tmp/object.txt && cat /tmp/object.txt"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Ray cluster management"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This workspace is connected to a Ray cluster. Click on the resources bar on the top right corner of the screen to open the cluster control panel. This panel shows a summary of Ray resource utilization, and you can use this panel to configure the cluster resources.\n",
+    "\n",
+    "<img src=\"assets/resource-panel.png\" height=400px/>\n",
+    "\n",
+    "### Configuring the Workspace node\n",
+    "\n",
+    "The workspace node is the machine this notebook is running inside. You may wish to change the instance type of the workspace node specifically, e.g., to increase the available memory or add a GPU. Click the pencil icon in order to change the workspace node. Note that changing the workspace node will restart the workspace IDE.\n",
+    "\n",
+    "<img src=\"assets/edit-workspace-node.png\" height=300px/>\n",
+    "<img src=\"assets/edit-workspace-dialog.png\" width=400px/>\n",
+    "\n",
+    "### Adding worker nodes\n",
+    "\n",
+    "To parallelize beyond the resources available to the workspace node, add additional worker nodes to the Ray cluster. Click \"Add a node type\" to add a number of nodes of a certain type to the cluster. While most use cases only require a single worker node type, you can add multiple distinct node types (e.g., high-CPU and GPU nodes) to the workspace as well.\n",
+    "\n",
+    "<img src=\"assets/add-node-type.png\" height=300px/>\n",
+    "<img src=\"assets/add-node-dialog.png\" height=300px/>\n",
+    "\n",
+    "### Using \"Auto\" workers mode\n",
+    "\n",
+    "To let Ray automatically select what kind of worker nodes to add to the cluster, check the \"Automatically select workers\" box. Ray will try to autoscale cluster workers to balance cost and performance. In auto mode, you cannot configure worker node types, but the resources panel will show which node types have been launched.\n",
+    "\n",
+    "We recommend using auto mode if you do not have specific cluster requirements, and are ok with waiting for the autoscaler to add nodes on-demand to the cluster."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Monitoring Ray applications\n",
+    "\n",
+    "In this section, we'll author a simple Ray python script and go over the tools available to monitor its execution. Let's take the opportunity to create a `my_app.py` file in the `my_repo` git repo you created earlier.\n",
+    "\n",
+    "You can click on the \"File Explorer\" in the left pane of VSCode to create the new file. Copy paste the following program into the file:\n",
```

Comment:
Why don't we let them run it in notebook directly?

Reply:
Mainly I wanted to teach them how to run file-based Ray projects here as well, since that is a big use case for workspaces.